allennlp/predictors/semantic_role_labeler.py

```diff
@@ -198,16 +231,17 @@ def predict_json(self, inputs: JsonDict) -> JsonDict:
         if not instances:
             return sanitize({"verbs": [], "words": self._tokenizer.split_words(inputs["sentence"])})
 
-        outputs = self._model.forward_on_instances(instances)
+        return self.predict_instances(instances)
 
-        results = {"verbs": [], "words": outputs[0]["words"]}
-        for output in outputs:
-            tags = output['tags']
-            description = self.make_srl_string(output["words"], tags)
-            results["verbs"].append({
-                    "verb": output["verb"],
-                    "description": description,
-                    "tags": tags,
-            })
+    def predict_words_list(self, words_list: List[str]) -> JsonDict:
+        """
+        Create an instance list of works document, for skipping tokenization when that
+        information already exist for the user
+        """
+        tokens = self._tokenizer.tokens_from_list(words_list)
```

Comment:
You can't rely on this method actually existing, because you didn't create the method on the base class (I'm a little surprised that mypy didn't catch this; maybe because we're playing a little loose with the tokenizers inside of a predictor already...).  But it's a lot easier than this: just do `tokens = [Token(word) for word in words]`.

Reply:
Did you mean here I should just add ``tokens = [Token(word) for word in words]`` at WordSplitter base class?

Reply:
Instead of the line that you have, you should have `tokens = [Token(word) for word in words]`.  There's no need for any extra methods on `WordSplitter` for any of what you're implementing.

Reply:
Spacy POS tagging is required, I've tried to follow the class logic were all spacy pipeline is under the hood via class ``SpacyWordSplitter``

Reply:
That's especially non-obvious from the diff I was looking at.  And now that I look at the things that were hidden, I understand why mypy didn't catch this - I didn't realize that we specifically instantiated a `SpacyWordSplitter` here.  I'd just do what you did in the coref predictor - call `self._tokenizer.tokenizer.tokens_from_list()`, or whatever the method is.